addons/project_mail_plugin/controllers/project_client.py

```diff
@@ -0,0 +1,41 @@
+
+from odoo import http
+from odoo.http import request
+from odoo.tools import html2plaintext
+
+
+class ProjectClient(http.Controller):
+    @http.route('/mail_plugin/project/search', type='json', auth='outlook', cors="*")
+    def projects_search(self, search_term, limit=5):
+
+        projects = request.env['project.project'].search([('name', 'ilike', search_term)], limit=limit)
+
+        return [
+            {
+                'name': project.name,
+                'project_id': project.id,
+            }
+            for project in projects
+        ]
+
+    @http.route('/mail_plugin/task/create', type='json', auth='outlook', cors="*")
+    def task_create(self, email_subject, email_body, project_id, partner_id):
+
+        partner = request.env['res.partner'].browse(partner_id).exists()
+        if not partner:
+            return {'error': 'partner_not_found'}
+
+        record = request.env['project.task'].create({
+            'name': html2plaintext(email_subject),
+            'partner_id': partner_id,
+            'description': html2plaintext(email_body),
```

Comment:
Looks like the project.task description field supports HTML, wouldn't it be nice to keep HTML here?
Maybe some kind of sanitizing would be necessary though, to double-check.

Reply:
Yes, I think we can use HTML sanitize.